### PR TITLE
checker: fix the too many orphan peers cannot be removed

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/6573 

### What is changed and how does it work?

```commit-message
rule-checker: fix the too many orphan peers that cannot be removed
- let the health peer can be removed once there exist redundant
```

### Check List

Tests

- Unit test

### Release note

```release-note
Fix the issue too many peers/learners cannot be removed
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

File: pkg/schedule/checker/rule_checker_test.go [hunk 1]
```diff
@@ -168,6 +168,39 @@ func (suite *ruleCheckerTestSuite) TestFixOrphanPeers() {
...
 }
 
+func (suite *ruleCheckerTestSuite) TestFixToManyOrphanPeers() {
+	suite.cluster.AddLeaderStore(1, 1)
+	suite.cluster.AddLeaderStore(2, 1)
+	suite.cluster.AddLeaderStore(3, 1)
+	suite.cluster.AddLeaderStore(4, 1)
+	suite.cluster.AddLeaderStore(5, 1)
+	suite.cluster.AddLeaderStore(6, 1)
+	suite.cluster.AddRegionWithLearner(1, 1, []uint64{2, 3}, []uint64{4, 5, 6})
```
Comment (CabinfeverB) [inline pkg/schedule/checker/rule_checker_test.go:178]: This comment should have nothing to do with this PR. But I want to know what happen if here is `suite.cluster.AddRegionWithLearner(1, 1, []uint64{2, 3}, []uint64{4})`, and other behaviors are same.

Comment (CabinfeverB) [reply pkg/schedule/checker/rule_checker_test.go:178]: It seems not return operator

**Review by CabinfeverB — APPROVED** [review]

File: pkg/schedule/checker/rule_checker_test.go (hunk 1 above)
Comment (nolouch) [reply pkg/schedule/checker/rule_checker_test.go:178]: yes, because need len(`OrphanPeers`) >=2

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**